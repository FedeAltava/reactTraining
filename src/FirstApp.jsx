import PropTypes from 'prop-types'
export const FirstApp = ({title}) => {
  return (
    <>
        <h1>{title}</h1>
        <p>el fede</p>
    </>
  )
}

FirstApp.PropTypes = {
  title:PropTypes.string
}
