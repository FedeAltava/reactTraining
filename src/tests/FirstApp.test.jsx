
import { describe, test, expect} from "@jest/globals";
import { render } from "@testing-library/react";
import { FirstApp } from "../FirstApp";

describe('FirstApp', ()=>{
    test('should match with snapshot',()=>{
        const title = 'eyyyyyyy';
        const subtitle = 'yooooo';
        const name = 'ELPepe';
        const {container} = render(<FirstApp title={title} subtitle={subtitle} name={name}/>);

        expect(container).toMatchSnapshot();

    });
    test('should show title as h1',()=>{
        const title = 'eyyyyyyy';
        const {container, getByText} = render(<FirstApp title={title} />);
        const h1 = container.querySelector('h1');
        getByText(title)
        expect(h1.innerHTML).toContain(title);

    });

    test('should show the subtitle by props',()=>{
        const title = 'fede';
        const subtitle = 'yooooo';
        const {getAllByText} = render(<FirstApp title={title} subtitle={subtitle} />);
        expect(getAllByText(subtitle)).toBeTruthy();

    });
});
