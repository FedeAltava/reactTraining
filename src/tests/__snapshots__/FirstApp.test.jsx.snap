// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp should match with snapshot 1`] = `
<div>
  <h1>
    eyyyyyyy
  </h1>
  <h2>
    ELPepe
  </h2>
  <h3>
    yooooo
  </h3>
</div>
`;
